alf/environments/alf_wrappers.py

```diff
@@ -465,3 +481,285 @@ def reward_spec(self):
+            }))
+        self._zero_curriculum_info = self._add_task_names({
+            'curriculum_task_count': [torch.zeros(batch_size, device='cpu')] *
+                                     num_tasks,
+            'curriculum_task_score': [torch.zeros(batch_size, device='cpu')] *
+                                     num_tasks,
+            'curriculum_task_prob': [torch.zeros(batch_size, device='cpu')] *
+                                    num_tasks
+        })
+        self._progress_favor = progress_favor
+        self._current_score_update_rate = current_score_update_rate
+        self._past_score_update_rate = past_score_update_rate
+        self._warmup_period = warmup_period * num_tasks
+        self._scale = math.log(progress_favor)
+        self._total_count = 0
+        self._current_scores = torch.zeros(num_tasks, device='cpu')
+        self._past_scores = torch.zeros(num_tasks, device='cpu')
+        self._task_probs = torch.ones(num_tasks, device='cpu') / num_tasks
+        self._task_counts = torch.zeros(num_tasks, device='cpu')
+
+        self._current_task_ids = self._sample_tasks(batch_size)
+
+    def _add_task_names(self, info):
+        for k, v in info.items():
+            info[k] = dict(zip(self._task_names, v))
+        return info
+
+    def _sample_tasks(self, num_samples):
+        return torch.multinomial(
+            self._task_probs, num_samples=num_samples, replacement=True)
+
+    def _update_curriculum(self, task_ids, task_scores):
+        for task_id, task_score in zip(task_ids, task_scores):
+            self._total_count += 1
+            self._task_counts[task_id] += 1
+            self._current_scores[
+                task_id] += self._current_score_update_rate * (
+                    task_score - self._current_scores[task_id])
+            self._past_scores[task_id] += self._past_score_update_rate * (
+                task_score - self._past_scores[task_id])
+
+        progresses = (self._current_scores - self._past_scores).relu()
+        max_progress = progresses.max()
+        progresses = progresses / (max_progress + 1e-30)
+        # Gradually increase scale from 0 to self._scale so that we tend to do
+        # random smapling of the environments initially
+        scale = self._scale * min(1, self._total_count / self._warmup_period)
+        self._task_probs = F.softmax(scale * progresses, dim=0)
+
+    def env_info_spec(self):
+        return self._env_info_spec
+
+    def action_spec(self):
+        return self._action_spec
+
+    def _reset(self):
+        time_step = self._env.reset()
+        info = copy.copy(time_step.env_info)
+        info.update(self._zero_curriculum_info)
+        return time_step._replace(
+            env_info=info, prev_action=time_step.prev_action['action'])
+
+    def _step(self, action):
+        time_step = self._env.step(
+            OrderedDict(task_id=self._current_task_ids, action=action))
+        task_ids = time_step.prev_action['task_id']
```

Comment:
This should just be self._current_task_ids

Reply:
Changed